python_modules/dagster-graphql/dagster_graphql/schema/backfill.py

```diff
@@ -525,13 +525,21 @@ def resolve_runStatus(self, _graphene_info: ResolveInfo) -> GrapheneRunStatus:
         return GrapheneBulkActionStatus(self.status).to_dagster_run_status()
 
     def resolve_endTimestamp(self, graphene_info: ResolveInfo) -> Optional[float]:
+        if self._backfill_job.backfill_end_timestamp is not None:
+            return self._backfill_job.backfill_end_timestamp
         if self._backfill_job.status == BulkActionStatus.REQUESTED:
             # if it's still in progress then there is no end time
             return None
         records = self._get_records(graphene_info)
+        if len(records) == 0:
+            # backfill was moved to a terminal state before any runs were launched. We cannot
```

Comment:
also noticed in the UI that if a backfill ends before it launches any runs, the duration of the backfill continuously increases even though the backfill is completed. This should fix that issue. It'll show a duration of 0 seconds, but that seems preferable to a constantly growing amount of time